Reqnroll/Bindings/BindingFactory.cs

```diff
@@ -25,8 +25,8 @@ public IStepDefinitionBindingBuilder CreateStepDefinitionBindingBuilder(StepDefi
     }
 
     public IStepArgumentTransformationBinding CreateStepArgumentTransformation(string regexString,
-        IBindingMethod bindingMethod, string parameterTypeName = null)
+        IBindingMethod bindingMethod, string parameterTypeName = null, int order = default)
```

Comment:
There are two comments here.

Comment 1)

I would make the default order value a bit more explicit: create a const value somewhere (maybe in the `StepArgumentTransformationBinding` class) with the default value (of `0`) and use that everywhere, where you use `default` now. That would make it more explicit that the decision for the `0` was not because that is the default value of int, but because we wanted to use `0` as a default order (but see the other comment below).

Comment 2)

We should rethink, what is a good default value for the Order. Obviously we should encourage people to use `Order` for all transformations where the order matters, but this will not be followed in all cases and sometimes it is not possible (you want to extend the trafo of someone else that comes from a shared code). 

So the question is that if I receive an trafo without an order and I would like to extend (ie "override") it, what should I do? If I set `Order = 42` on mine that will run *after* the inherited one, because we associated `0` to that. So if I really want to override it, I need to use a negative number, like `Order = -1`. Maybe not that nice, but understandable. 

For hooks, we use Order = 10000 as default, which was decided by the Cucumber team and we just kept it for compatibility. We can consider to use that as a default for trafos as well, which would have the advantage of 

1. would be consistent with the hook default
2. would make it easier to "override" any trafo without an order (because 10000 is big enough, so every "meaningful" order number will be smaller anyway).

The disadvantage on the other hand would be that it introduces a magic number.

What do you think?

Reply:
Firstly, thanks for your feedback!

Personally, I try to avoid using magic numbers in the code because it always produces the question `why this number?`. There could be cases where someone intends to declare global step argument transformations with an order set to `1000` (which for many projects is quite enough already), and still these transformations will be executed before those without a specified order. 

If we're not tied to the existing implementation of hook order, I would prefer to stick with 0 as the default. However, since the similar implementation is already in place and it's quite subjective, let's maintain consistency across the solution.

----

I've created the constant value in `StepArgumentTransformationAttribute` instead of `StepArgumentTransformationBinding` aligning with the approach used in `HookAttribute`. The default value is 10000.